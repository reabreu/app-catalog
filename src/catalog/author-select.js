import React from "react";
import { AuthSelect, AuthOption } from "./styles";

export const AuthorSelect = ({ options, onChangeHandler }) => (
  <AuthSelect onChange={onChangeHandler}>
    <AuthOption value={""}>Select an author</AuthOption>
    {options.map((author) => (
      <AuthOption key={author} value={author}>
        {author}
      </AuthOption>
    ))}
  </AuthSelect>
);
