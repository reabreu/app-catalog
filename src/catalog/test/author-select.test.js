import React from "react";
import { render, fireEvent, wait, cleanup } from "@testing-library/react";
import { AuthorSelect } from "../author-select";

afterEach(cleanup);

describe("<Search /> spec", () => {
  it("Renders a select with provided options", () => {
    const { asFragment } = render(
      <AuthorSelect options={[1, 2, 3]} onChangeHandler={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Calls on Change for onChangeHandler method", async () => {
    const onChangeHandler = jest.fn();

    const { getByRole } = render(
      <AuthorSelect options={[1, 2, 3]} onChangeHandler={onChangeHandler} />
    );

    const select = getByRole("combobox");

    fireEvent.change(select, { target: { value: 1 } });

    await wait(() => expect(onChangeHandler).toHaveBeenCalled(), {
      timeout: 300,
    });
  });
});
