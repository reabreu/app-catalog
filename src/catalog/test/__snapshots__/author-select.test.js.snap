// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> spec Renders a select with provided options 1`] = `
<DocumentFragment>
  <select
    class="sc-fzozJi iKyCId"
  >
    <option
      class="sc-fzoLsD kNEntU"
      value=""
    >
      Select an author
    </option>
    <option
      class="sc-fzoLsD kNEntU"
      value="1"
    >
      1
    </option>
    <option
      class="sc-fzoLsD kNEntU"
      value="2"
    >
      2
    </option>
    <option
      class="sc-fzoLsD kNEntU"
      value="3"
    >
      3
    </option>
  </select>
</DocumentFragment>
`;
